src/Service/CdpClient.php

```diff
@@ -30,18 +29,31 @@ class CdpClient
      */
     private $apiKey;
 
+    /**
+     * The serializer.
+     *
+     * @var \Symfony\Component\Serializer\Serializer
+     */
+    private $serializer;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $cache;
+
     /**
      * CdpClient constructor.
      *
      * @param string $url
      * @param string $apiKey
      */
-    public function __construct(string $url, string $apiKey)
+    public function __construct(string $url, string $apiKey, AdapterInterface $adapter)
```

Comment:
This, unfortunately, breaks in drupal due to the way they have implemented their cache layer. I already tried to do it. I made a wrapper for the caching